api/scpca_portal/models/dataset.py

```diff
@@ -509,7 +521,7 @@ def spatial_projects(self) -> Iterable[Project]:
             for project_id, project_options in self.data.items()
             if project_options.get(Modalities.SPATIAL, [])
         ]:
-            return self.projects.filter(scpca_id__in=project_ids)
+            return self.projects.filter(has_spatial_data=True, scpca_id__in=project_ids)
```

Comment:
Let's add docstrings that clarify that this returns true or false for projects selected for the type of data